### ISSUE TITLE
0.6.16 BUG with IAM Role Attribute Mismatch

### ISSUE THREAD

**Issue by eedwardsdisco** [hashicorp/terraform#6834]
### Terraform Version

0.6.16
### Affected Resource(s)

Please list the resources as a list, for example:
- aws_iam_role
- aws_iam_instance_profile
### Terraform Configuration Files

https://gist.github.com/eedwardsdisco/5706ca2000a9adb4caf739b371e7b8b9
### Debug Output

https://gist.github.com/eedwardsdisco/4e84559902baf96d31dbc2a19cbca904
### Panic Output

https://gist.github.com/eedwardsdisco/419cb65ae7e497130f17c59d79d177e4
### Expected Behavior

terraform apply succeeds
### Actual Behavior

terraform apply fails and exits on a bug
### Steps to Reproduce

Please list the steps required to reproduce the issue, for example:
1. `terraform apply`
### Important Factoids

succeeds if I run apply again after the fact

**Comment by eedwardsdisco** [comment]
possibly related to #6788 ?

**Comment by eedwardsdisco** [comment]
possibly related to #6806 ?

**Comment by eedwardsdisco** [comment]
Here's a breakdown of the diff.

Seems to be an issue with the roles.X field, where the plan phase has a different role id than the apply phase?

https://gist.github.com/eedwardsdisco/0163b5de916cdf4307d9084c28a55a73

**Comment by eedwardsdisco** [comment]
I managed to get past it by changing from using a rendered template output for the name of the policy

```
name = "${template_file.cluster_name.rendered}-opsworks-service"
```

I changed the above (and all other references to this template) to instead be the value that was getting calculated in the template itself.

I suspect there is a problem with the plan phase and rendering template output.

e.g. sounds related to #5462 in that I've also seen the 'ERROR PARSING JSON' problem on PLAN but not on APPLY (when the JSON comes from a template output).
